posthog/api/sharing.py

```
@@ -98,8 +98,13 @@ def export_asset_for_opengraph(resource: SharingConfiguration) -> ExportedAsset
         context={"team_id": cast(Team, resource.team).pk},
     )
     serializer.is_valid(raise_exception=True)
-    export_asset = serializer.synthetic_create("opengraph image")
-    return export_asset
+
+    try:
```

I'm not sure I understand the logic behind this change - in this case we wouldn't capture the exception on the export itself? Is that what we want here for some reason?

---

since we now raise the exception instead of silently logging, the [test_sharing_refresh_with_personal_api_key](https://github.com/PostHog/posthog/blob/1e3813a2f8d762893687946400f90fe96c555742/posthog/api/test/test_personal_api_keys.py#L545) test was failing because when you initially save a sharing, it tries to screenshot it, in the screenshot process it opens the sharing page, which includes a reference to opengraph image, but it doesn't have a PNG generated yet so it failed, and to take a PNG you'd need a valid sharing saved, so that kind of created a circular dependency.